src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs

```diff
@@ -424,5 +424,10 @@ public static void Interrupt(Thread thread)
                 s_lock.Release();
             }
         }
+
+        public static void OnThreadExiting(Thread thread)
```

Comment:
I think Thread.Mono.cs is more approprivate place for this helper method.

Reply:
This same problem would have to be addressed in CoreCLR, right? Happy to move it though.

Reply:
Yes, but I am not sure whether we would address it the same way. We can always move it if necessary.

In any case, I think it should be on `Thread` so that it can be used for any thread exit handling even outside WaitSubsystem.